### PR TITLE
Port the dlisio target to a C++ API

### PR DESCRIPTION
**Pull request by ErlendHaa**

### CODE REVIEW

**Review by achaikou — COMMENTED** [review]
Looks fine to me with my non existing knowledge of upgrading C to C++ :D 
:+1: 

(And now Travis seems to be affected by the Docker limit... And it's not even bigendian job where we actually use Docker, but multibuild. Bad if it will happen during release... 
We might have to authorize then.
But maybe if we pretend good enough that it's not our problem, it will go away by itself...?...)

File: lib/include/dlisio/types.hpp [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+/*
+ * get the size (in bytes) of a particular data type. Expects a DLIS_UNORM or
+ * similar type code.
+ *
+ * Returns a negative value passed an invalid type code.
+ */
+int sizeof_type(int);
+
+constexpr int FSHORT = 1;  // Low precision floating point
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply lib/include/dlisio/types.hpp:178]: yes, exactly that 👍

File: lib/include/dlisio/types.hpp [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+/*
+ * A family of primitive functions to parse the data types specified by RP66.
+ * They all return a pointer to the first character NOT consumed by calling
+ * this function.
+ *
+ * If the output argument is NULL it is not written, but distance is still
+ * calculated. This is necessary for variable-length strings to ensure a large
+ * enough buffer. Strings will NOT be null terminated
+ */
+
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply lib/include/dlisio/types.hpp:16]: Then we leave it 😄

File: lib/include/dlisio/types.hpp [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+const char* ident_frombytes(const char*, int32_t* len, char* out);
+const char* ascii_frombytes(const char*, int32_t* len, char* out);
+
+constexpr int tz_lst = 0; // local standard
+constexpr int tz_dst = 1; // local daylight savings
+constexpr int tz_gmt = 2; // greenwich mean time
+
+constexpr int YEAR_ZERO = 1900;
+int dlis_year_frombytes( int );
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply lib/include/dlisio/types.hpp]: Removed the entire function

File: lib/include/dlisio/dlisio.hpp [hunk 4]
```diff
@@ -50,67 +46,61 @@ int dlis_sul( const char* xs,
...
-int dlis_tapemark(const char* buffer, int size);
+int tapemark(const char* buffer, int size);
 
-#define DLIS_VRL_SIZE 4
-DLISIO_API
-int dlis_vrl( const char* xs,
-              int* len,
-              int* version );
+constexpr int VRL_SIZE = 4;
+int vrl( const char* xs,
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply lib/include/dlisio/dlisio.hpp:68]: Gonna keep

File: lib/include/dlisio/dlisio.hpp [hunk 5]
```diff
@@ -1,29 +1,25 @@
-#ifndef DLISIO_H
-#define DLISIO_H
+#ifndef DLISIO_HPP
```
Comment (achaikou) [inline lib/include/dlisio/dlisio.hpp:1]: Search this file for "DLIS_". There are couple of things in the comments

There is also one in ext/types.hpp

**Pull request closed without merging**